CustomApps/lyrics-plus/ProviderMusixmatch.js

```diff
@@ -45,6 +45,61 @@ const ProviderMusixmatch = (function () {
 		return body;
 	}
 
+	async function getKaraoke(body) {
+		const meta = body?.["matcher.track.get"]?.message?.body;
+		if (!meta) {
+			return null;
+		}
+
+		if (!meta.track.has_richsync || meta.track.instrumental) {
+			return null;
+		}
+
+		const baseURL = `https://apic-desktop.musixmatch.com/ws/1.1/track.richsync.get?format=json&subtitle_format=mxm&app_id=web-desktop-app-v1.0&`;
+
+		const params = {
+			subtitle_length: meta.track.track_length,
+			commontrack_id: meta.track.commontrack_id,
+			usertoken: CONFIG.providers.musixmatch.token
+		};
+
+		const finalURL =
+			baseURL +
+			Object.keys(params)
+				.map(key => key + "=" + encodeURIComponent(params[key]))
+				.join("&");
+
+		let result = await CosmosAsync.get(finalURL, null, headers);
+
+		if (result.message.header.status_code != 200) {
+			return null;
+		}
+
+		result = result.message.body;
+
+		const parsedKaraoke = JSON.parse(result.richsync.richsync_body).map(e => {
+			const words = e.l.map((f, i, a) => {
```

Comment:
Is it possible to use descriptive variable names here instead of single characters like this?
We would appreciate something that can be easily understood by whomever had the knowledge.

Reply:
Corrected with https://github.com/spicetify/spicetify-cli/pull/2248/commits/8343197bb18c5dcf3ab9536414e1d44ffd8f5219. How about something like this?

Reply:
Looks pretty good to me.